.github/workflows/slurm_job_scripts/multinode_pytest.sub

```diff
@@ -34,7 +34,7 @@ read -r -d '' cmd <<EOF
 date \
 && pytest -v  -s --continue-on-collection-errors \
     --junit-xml=/workspace/outputs/junit_output_\${SLURM_PROCID}.xml \
-    /workspace/tests/distributed_multinode_test.py
+    /workspace/tests/distributed_gpu_multiprocess_test.py
```

Comment:
Well having 2 tests called distrbuted_test.py and distributed_multinode_test.py is confusing. Which platform is distributed_multinode_test.py on? (distributed_test.py is also not a good name, so that should also be changed).

What's a node anyways? In JAX's terminology, we have processes i.e. hosts (which is what you mean by node?). FYI: hosts has been deprecated in favor of process!

I was against having 2 files to begin with. I don't see any problem just merging them right now. But its fine for iterating quickly and then merging them.

Reply:
I see the end test as having 1 file with name `distributed_gpu_multiprocess_test.py` or just `gpu_multiprocess_test.py` (this is more consise).

Reply:
I agree with you about the issues with having two separate files for testing similar thing (`jax.distributed.initialize`). The only difference is that with `distributed_multinode_tests.py` these processes run on different nodes (hosts with a set of GPUs each, they're called "nodes" in [HPC terms](https://www.hpc.iastate.edu/guides/introduction-to-hpc-clusters/what-is-an-hpc-cluster)) and a JAX process runs on each node (I'm also working on adding a test that will run one process per GPU). Not imposing any terms on JAX, I just found it easy to begin with and will work with you to align with JAX terminology. 

I support having `gpu_multiprocess_test.py` as the end goal. I can work towards making changes towards that in the next PR (tentatively next week along with another set of tests that I'm planning to add, and with GPU jaxlib nightly). Does it sound good?

Reply:
Also something to check out: https://jax.readthedocs.io/en/latest/multi_process.html

> hosts with a set of GPUs each

That's fine. Its still a process for JAX. Looking at it from the API perspective, JAX only has `jax.process_index()`, `jax.process_count()`, etc. So its good to stick to those terms even when naming files otherwise it becomes confusing. Its explained more in the above doc.

> I support having gpu_multiprocess_test.py as the end goal. I can work towards making changes towards that in the next PR (tentatively next week along with another set of tests that I'm planning to add, and with GPU jaxlib nightly). Does it sound good?

Yeah sounds good to me as long as we clean this up sooner rather than later. So the plan is to have just 1 file called: `gpu_multiprocess_test.py` and removing `distributed_test.py` and the file you added right? 

So you can have 1 class:

```
class MultiprocessGpuTest():

  def ...
  
  @unittest.SkipIf(...)
  def slurm_test():
```

Does this look good to you?

Also might be worth cleaning this up first before sending CLs adding more test as they are both orthogonal to each other and makes reviewing easier.

Reply:
> Also might be worth cleaning this up first before sending CLs adding more test as they are both orthogonal to each other and makes reviewing easier.

Makes sense.



> That's fine. Its still a process for JAX. Looking at it from the API perspective, JAX only has `jax.process_index()`, `jax.process_count()`, etc. So its good to stick to those terms even when naming files otherwise it becomes confusing. Its explained more in the above doc.

Sure, I agree sticking to those terms would avoid filenames become confusing.



> Yeah sounds good to me as long as we clean this up sooner rather than later. So the plan is to have just 1 file called: `gpu_multiprocess_test.py` and removing `distributed_test.py` and the file you added right?

Sure, having one file and one class as you suggested sounds good. Regarding naming the file - how do we name tests currently? Is it based on module name (`jax.distributed` in this case and so `distributed_test.py`?) or what it actually does (like `gpu_multiprocess_test.py` which makes it specific for platform)?

Reply:
I like `gpu_multiprocess_test.py`. Its good to be clear about the platform because we might have `tpu_multiprocess_test.py`  in the future too. `distributed_test.py` is not idea too, so this cleanup would be nice :)

Reply:
Sounds good, let me do that clean up and get back with a PR. Thanks for the useful info! :)

Reply:
I guess it's okay to close this PR then?

Reply:
Sounds good! Thanks!